### PR TITLE
Feat/delete account

### PR DESCRIPTION
**Pull request by ivan-dalmet**
Add the flow to allows users to delete their account

### CODE REVIEW

**Comment by ntatoud** [comment]
[Capture vidéo du 23-05-2024 09:18:57.webm](https://github.com/BearStudio/start-ui-web/assets/135032615/75bbb87f-78e4-48f9-b351-07665dde2966)

Above ⬆️ , a demo of how the flow looks like.

**Review by DecampsRenan — APPROVED** [review]

File: src/features/account/AccountDeleteVerificationCodeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  const onClose = () => {
+    setSearchParams({
+      [SEARCH_PARAM_VERIFY_EMAIL]: null,
+      token: null,
+      attempts: null,
+    });
+  };
+
+  const form = useForm<{ code: string }>({
```
Comment (yoannfleurydev) [inline src/features/account/AccountDeleteVerificationCodeModal.tsx:52]: We don't infer from the zod schema here?

File: src/server/routers/account.tsx [hunk 2]
```diff
@@ -204,6 +201,86 @@ export const accountRouter = createTRPCRouter({
...
       await deleteUsedCode({ ctx, token: verificationToken.token });
 
+      return user;
+    }),
+
+  deleteRequest: protectedProcedure()
+    .meta({
+      openapi: {
+        method: 'POST',
+        path: '/account/delete/request',
```
Comment (yoannfleurydev) [inline src/server/routers/account.tsx:211]: should probably be `/account/me/delete/request`

File: src/server/routers/account.tsx [hunk 3]
```diff
@@ -204,6 +201,86 @@ export const accountRouter = createTRPCRouter({
...
+      return {
+        token,
+      };
+    }),
+
+  deleteValidate: protectedProcedure()
+    .meta({
+      openapi: {
+        method: 'DELETE',
+        path: '/account/delete/validate/',
```
Comment (yoannfleurydev) [inline src/server/routers/account.tsx:262]: same comment here `/account/me/delete/validate`

**Comment by yoannfleurydev** [comment]
Overall OK to me, can be shipped liked that, only some suggestions

File: src/features/account/AccountDeleteVerificationCodeModal.tsx (hunk 1 above)
Comment (ivan-dalmet) [reply src/features/account/AccountDeleteVerificationCodeModal.tsx:52]: Yes we should, I will fix that

File: src/server/routers/account.tsx (hunk 2 above)
Comment (ivan-dalmet) [reply src/server/routers/account.tsx:211]: we don't have the /me for update email "path: '/account/update-email/'," so 3 solutions :
- Remove /me for delete
- Add /me for update email
- Remove /me for all routes (/account/me -> /account)

2 and 3 will change the routes for Start UI Native

Comment (yoannfleurydev) [reply src/server/routers/account.tsx:211]: > Remove /me for delete

OK to me

**Pull request merged**